src/components/shared/Fields/Form/HookForm.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import {
+  useForm,
+  UseFormProps,
+  FormProvider,
+  UseFormReturn,
+  SubmitHandler,
+  SubmitErrorHandler,
+} from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Schema } from 'yup';
+
+const displayName = 'HookForm';
+
+export interface HookFormProps<FormData extends Record<string, any>> {
+  children: ((props: UseFormReturn) => React.ReactNode) | React.ReactNode;
+  onSubmit: SubmitHandler<FormData>;
+  onError?: SubmitErrorHandler<FormData>;
+  validationSchema?: Schema<FormData>;
+  defaultValues?: UseFormProps['defaultValues'];
+  mode?: UseFormProps['mode'];
+  options?: UseFormProps;
+}
+
+const HookForm = <FormData extends Record<string, any>>({
+  children,
+  defaultValues,
+  mode = 'onTouched',
+  onSubmit,
+  onError,
+  options,
+  validationSchema,
```

Comment:
Mode (i.e. validation strategy), validation schema and default values actually belong in the options object passed to the `useForm`. However, seeing as they're likely some of the more used options, I've pull them out and added them as props for convenience.

Reply:
Potentially if you have a large form and you don't want it to validate on change before it's been submitted for performance reasons. This is the hook form default ('onSubmit'). 

But I think generally it'll be 'onTouched' since this was the standard in the dapp.